src/solver.jl

```diff
@@ -32,15 +32,16 @@ function POMDPs.solve(sol::CWorldSolver, w::CWorld)
                 newdata[i] = 0.0
             else
                 best_Qsum = -Inf
-                for a in actions(w, s)
+                for a in actions(w)
                     Qsum = 0.0
                     for j in 1:sol.m
-                        sp, r = @gen(:sp, :r)(w, s, a, sol.rng)
-                        Qsum += r + discount(w)*evaluate(val, sp)
+                        sp = rand(transition(w, s, a, sol.rng))
+                        r = reward(w, s, a, sp)
+                        Qsum += r + discount(w) * evaluate(val, sp)
                     end
                     best_Qsum = max(best_Qsum, Qsum)
                 end
-                newdata[i] = best_Qsum/sol.m
+                newdata[i] = best_Qsum / sol.m
```

Comment:
These changes make the solver less general. Is there a reason why you would like to commit these changes? I recommend reverting them.